src/components/CodeEditor/CodeEditor.jsx

```diff
@@ -26,24 +34,73 @@ export default function CodeEditor(props) {
     setChecked(event.target.checked);
   }
 
-  useEffect(() => {
-    checked ? setTheme("vs-dark") : setTheme("vs-light");
-  }, [checked]);
+  function checkError(data) {
+    let errorList = [];
+    let hasError = false;
+    data.errors.forEach((error, index) => {
+      if (data.errors[`${index}`].severity === "error" || data.errors[`${index}`].errorCode === "3420")
+      {
+        hasError = true;
+        var errorInfo = {
+          ERROR: `Compilation failed: ${data.errors[`${index}`].message} (type: ${data.errors[`${index}`].type}, code: ${data.errors[`${index}`].errorCode}).`
+        }
+        errorList.push(errorInfo);
+      }
+    })
+    if (hasError) {
+      return errorList;
+    }
+    return null;
+  }
 
-  async function compile() {
+  async function compile(value, event) {
     const response = await fetch("http://localhost:3001/compile", {
       headers: {
         "Content-Type": "application/json",
       },
       method: "POST",
       body: JSON.stringify({ value: editorRef.current.getValue() }),
     });
+
     const data = await response.json();
-    props.onCompile(data);
+    console.log(data)
+    const errorInfo = checkError(data);
+    if (errorInfo != null) {
+      props.onCompile(errorInfo)
+      return;
+    }
+    
+    // pretty scuffed, but works for now
+    let abiOutput = {
+      abi: [],
+    };
+    let contractName = "";
+    for (var cName in data.contracts["test.sol"]) {
+      contractName = cName;
+      abiOutput.abi.push(data.contracts["test.sol"][contractName].abi);
+    }
```

Comment:
`  const abiOutput = Object.keys(data.contracts["test.sol"]).map((key) => {
      return {
        name: key,
        abi: data.contracts["test.sol"][key].abi,
      };
    });`
    
 a more javascript way of doing the same thing, either way is fine.